linode/instance/schema_resource.go

```diff
@@ -67,26 +67,109 @@ func resourceDeviceDisk() *schema.Resource {
 	}
 }
 
-func resourceConfigInterface() *schema.Resource {
-	return &schema.Resource{
-		Schema: map[string]*schema.Schema{
-			"label": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "The unique label of this interface.",
-			},
-			"purpose": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "The purpose of this interface.",
+const (
+	onlyAllowedForVPCMsg  = "This attribute is only allowed for VPC interfaces."
+	onlyAllowedForVLANMsg = "This attribute is only allowed for VLAN interfaces."
+	requiredForVPCMsg     = "This attribute is required for VPC interfaces."
+	requiredForVLANMsg    = "This attribute is required for VLAN interfaces."
+)
+
+var InterfaceSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"purpose": {
+			Type:        schema.TypeString,
+			Description: "The type of interface.",
+			Required:    true,
+			ValidateDiagFunc: validation.ToDiagFunc(
+				validation.StringInSlice([]string{"public", "vlan", "vpc"}, true),
+			),
+		},
+		"ipam_address": {
+			Type: schema.TypeString,
+			Description: "This Network Interface's private IP address in " +
+				"Classless Inter-Domain Routing (CIDR) notation." +
+				onlyAllowedForVLANMsg,
+			Optional: true,
+		},
+		"label": {
+			Type: schema.TypeString,
+			Description: "The name of the VALN. " + requiredForVLANMsg +
+				" " + onlyAllowedForVLANMsg,
+			Optional: true,
+		},
+		"id": {
+			Type:        schema.TypeInt,
+			Description: "The ID of the interface.",
+			Computed:    true,
+		},
+		"subnet_id": {
+			Type: schema.TypeInt,
+			Description: "The ID of the subnet which the VPC interface is connected to." +
+				requiredForVPCMsg + onlyAllowedForVPCMsg,
+			Optional: true,
+		},
+		"vpc_id": {
+			Type: schema.TypeInt,
+			Description: "The ID of VPC of the subnet which the VPC " +
+				"interface is connected to.",
+			Computed: true,
+		},
+		"primary": {
+			Type: schema.TypeBool,
+			Description: "Whether the interface is the primary interface that should " +
+				"have the default route for this Linode.",
+			Optional: true,
+			Default:  false,
+		},
+		"active": {
+			Type:        schema.TypeBool,
+			Description: "Whether this interface is currently booted and active.",
+			Computed:    true,
+		},
+		"ip_ranges": {
+			Type:        schema.TypeList,
+			Description: "List of VPC IPs or IP ranges inside the VPC subnet.",
+			Optional:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+				ValidateDiagFunc: validation.AnyDiag(
+					helper.SDKv2ValidateIPv4Range,
+					helper.SDKv2ValidateIPv6Range,
+				),
```

Comment:
It looks like this field isn't being returned without the extra VPC features being enabled on your account. Do we want to include it anyway or would it make sense to split this out into a separate post-beta PR?

Reply:
I think we already include many things that needs extra beta in different repo, so I guess this is fine? Like this field is in Ansible, lol